tutorial/README.md

```diff
@@ -0,0 +1,824 @@
+```
+
+---
+
+If you need to fix a mistake the best way to do it is to clear out all of the resources and then apply them again with the correct versions.
+
+```sh
+# using the control plane VM
+azoam flight-tracker/* | kubectl delete -f -
+
+# after fixing the mistake
+azoam flight-tracker/* | kubectl apply -f -
+
+```
+
+
+### Application Developer: Authoring and registering components
+
+There is one UI microservice, four API microservices and one Postgres database.
+Each team is responsible for delivering the component schematic for their
+microservice. The Postgres database can either be run in-cluster or be consumed
+as a managed service from a cloud provider. 
+
+1. The **OAM Components** that are applied require the following information
+   about the app from the developers:
+
+    * The workload type, which dictates how the microservice is supposed to run.
+      In this example, all microservices are of type `ContainerizedWorkload`.
+      The Postgres database is of type `PostgreSQLInstance`.
+    * The container image and credentials. Developers are responsible at the
+      very least for authoring the Dockerfiles containing the dependencies in
+      order to build their runnable container. This example also expects an
+      image to be pushed to a registry although this can be handled by a
+      continuous integration system.
+    * Container ports that expose any ports that servers are listening to.
+    * Parameters that can be overridden by an operator at time of instantiation.
+
+2. Register the `Components`.
+
+    ```
+    kubectl apply -f tracker-managed-db-component.yaml
+    kubectl apply -f tracker-data-component.yaml
+    kubectl apply -f tracker-flights-component.yaml
+    kubectl apply -f tracker-quakes-component.yaml
+    kubectl apply -f tracker-weather-component.yaml
+    kubectl apply -f tracker-ui-component.yaml
+    ```
+
+*Note: if you prefer to run the Postgres database in-cluster using a
+`ContainerizedWorkload`, register the `tracker-db-component.yaml` instead of the
+`tracker-managed-db-component.yaml`.*
+
+### Application Operator: Instantiate application with appropriate configuration
+
+The application operator (this may or may not be different than the developer)
+tasks involve running the application with appropriate configurations.
+
+1. This **OAM ApplicationConfiguration** instantiates each of the components and
+   allows the operator to tune the following when running the components:
+
+    * Number of replicas for each component
+    * Values for any parameters that can be overridden in the components.
+
+2. Install the `ApplicationConfiguration`.
+
+    ```
+    kubectl create -f ApplicationConfiguration/tracker-app-config-managed.yaml
+    ```
+
+*Note: if you prefer to run the Postgres database in-cluster, install the
+`track-app-config.yaml` `ApplicationConfiguration` instead of
+`tracker-app-config-managed.yaml`.*
+
+3. Fine the external IP of the UI microservice
```

Comment:
Find

Reply:
This section was a copy-paste from the original flight tracker repo and shouldn't be here. I'm reworking this.